### ISSUE TITLE
Offset in user creationtime ?

### ISSUE THREAD

**Issue by cturbelin** [influenzanet/user-management-service#6]
https://github.com/influenzanet/user-management-service/blob/6d3b1b309110e4c85065f18175fa850958464247/pkg/grpc/service/user_management_endpoints.go#L65

What is the purpose of adding an offset to the creation time of a user has an offset ? The constant is set with an offset of 7 days which is a big offset.

https://github.com/influenzanet/user-management-service/blob/6d3b1b309110e4c85065f18175fa850958464247/pkg/grpc/service/constants.go#L14

The comment suggest is for user deletion ? But why adding it to the creation time (instead of at check time when to decide if user should be deleted) ?

**Comment by phev8** [comment]
Yes, this is to offset user deletion, when user was created by an admin user. That happens when users get invited to the system (e.g. migrated from the old platform). 
The motivation behind is:
- when users register themselves such offset is not used, since we assume, that if they start registration they are willing to finish the process as well. Also has to do with how long the tokens for finishing up the registration or verifying the email address.
- but when invited by the system, they receive the email, they need to find time to read it and take action... If we don't offset that somehow, their invited account would be already deleted until they open up the email probably.

This was the simplest way to achieve this. In the logs, the correct time is recorded, so you have the info if necessary, but from the system's perspective it provided a simple way to prevent the user to be deleted prematurely.

**Comment by cturbelin** [comment]
Ah ok, yes I just get there are 2 functions to create user : one from admin and from signup (we should probably look the opportunities to factorize this 2 functions they look very similar, maybe some steps could be shared).

I'll just add a comment to this two functions to clarify this ! Thanks
